### DISCUSSION TITLE
Could CC be employed in a code generator?

### DISCUSSION THREAD

**Discussion by rafalkrupinski** [cookiecutter/cookiecutter#1806] [Q&A]
Not sure if it's Q&A or Ideas, but could Cookiecutter be used in a code generator?

I've never used CC, just looked thru the documentation.

From what I've understood, CC focuses on initiating projects - creating singular files like pyproject or gitignore.

Could I feed it with a set of models to generate a set of files of the same kind? Like set of classes representing a SQL database tables or JSONSchema objects?

**Comment by kurtmckee** [comment]
I think that cookiecutter works on a fixed set of files and directories, so I don't think it's possible to take an arbitrary list of _x_ tables and generating separate files for each table and its schema.

Therefore I think you might be better served by writing custom code for this use case.
